static-parts/index-start.html

```diff
@@ -17,9 +17,16 @@
   </head>
   <script>
     const search = (node) => {
-      Array.from(node.parentElement.parentElement.getElementsByTagName('a')).forEach(x => x.style.display = x.getAttribute("href").toUpperCase().indexOf(node.value.toUpperCase()) + 1  ? '' : 'none');
+      Array.from(node.parentElement.parentElement.getElementsByTagName('a')).forEach(x => {
+        x.style.display = x.getAttribute("href").toUpperCase().indexOf(node.value.toUpperCase()) + 1  ? '' : 'none'
+      });
       var remove = false, childs = Array.from(node.parentElement.parentElement.children), toRemove = childs[0];
-      Array.from(node.parentElement.parentElement.children).forEach(x => x.tagName == 'H3' ? (toRemove.style.display = (remove ? 'none' : ''), toRemove = x, remove = true) : (x.style.display == '' ? remove = false : remove=remove));
+      Array.from(node.parentElement.parentElement.children).forEach((x, idx) => {
```

Comment:
there's a design flaw here. If you want to hide any empty categories, you merely need to get all children, filter based on if they want to display, and the forEach and check if the type of the element tag is H3 and if the next one is also H3, then set this one to hide.

Also Array.forEach has an API of `(item, index, sourceArray) => functionBody`

Reply:
Also if last element in the array and it is an H3 it should also cause it to hide.

As it stands what I'm reading right now you're relying on state information that is unneeded.

Reply:
```js
Array.from(node.parentElement.parentElement.children
.filter( x => !( x.tagName == 'a' && x.style.display == 'none' ) )
.forEach( ( element, index, source) => {
  element.style.display = element.tagName == 'H3' ? index + 1 == source.length ? 'none' : source[index + 1].tagName == 'H3' ? 'none'  : ''
})
```

Reply:
Okay the filter on my code is bad. I'm missing something...

Reply:
I cannot check this right now, but I trust that @skatcat31 is onto something from what I'm reading. Has the submitted PR been tested thoroughly? Is there a faster/easier way? I wouldn't know because I cannot easily test this right now.

Reply:
@skatcat31 At least on my local, your snippet does not work.

Reply:
To be honest, I would leave that like this. Take a look: JS takes **3.5ms** which takes about 15% of all loading time. 

![2017-12-23_15_26_47](https://user-images.githubusercontent.com/16862997/34320250-ba79e9de-e7f5-11e7-9ab6-28bd7568254c.png)

![2017-12-23_15_26_26](https://user-images.githubusercontent.com/16862997/34320249-b7e8d00e-e7f5-11e7-99fe-1d22f56f3563.png)

Reply:
@skatcat31 Your version takes even longer
![2017-12-23_15_43_02](https://user-images.githubusercontent.com/16862997/34320322-facc9868-e7f7-11e7-959a-9ee54f85a91c.png)